packages/request-client.js/src/http-metamask-data-access.ts

```diff
@@ -0,0 +1,306 @@
+import { Block } from '@requestnetwork/data-access';
+import { requestHashSubmitterArtifact } from '@requestnetwork/smart-contracts';
+import { DataAccessTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+import axios, { AxiosRequestConfig } from 'axios';
+import { ethers } from 'ethers';
+
+// Maximum number of retries to attempt when http requests to the Node fail
+const HTTP_REQUEST_MAX_RETRY = 3;
+
+// Delay between retry in ms
+const HTTP_REQUEST_RETRY_DELAY = 100;
+
+/**
+ * Exposes a Data-Access module over HTTP
+ */
+export default class HttpMetamaskDataAccess implements DataAccessTypes.IDataAccess {
+  /**
+   * Cache block persisted directly (in case the node did not have the time to retrieve it)
+   * (public for easier testing)
+   */
+  public cache: {
+    [channelId: string]: {
+      [ipfsHash: string]: { block: DataAccessTypes.IBlock; storageMeta: any } | null;
+    };
+  } = {};
+
+  /**
+   * Configuration that will be sent to axios for each request.
+   * We can also create a AxiosInstance with axios.create() but it dramatically complicates testing.
+   */
+  private axiosConfig: AxiosRequestConfig;
+
+  private submitterContract: ethers.Contract | undefined;
+  private provider: ethers.providers.JsonRpcProvider | ethers.providers.Web3Provider;
+  private networkName: string = '';
+
+  /**
+   * Creates an instance of HttpDataAccess.
+   * @param nodeConnectionConfig Configuration options to connect to the node. Follows Axios configuration format.
+   */
+  constructor(
+    {
+      nodeConnectionConfig,
+      web3,
+      urlProvider,
+    }: {
+      nodeConnectionConfig?: AxiosRequestConfig;
+      web3?: any;
+      urlProvider?: string;
+    } = {
+      nodeConnectionConfig: {},
+    },
+  ) {
+    urlProvider = urlProvider ? urlProvider : 'http://localhost:8545';
+
+    this.axiosConfig = Object.assign(
+      {
+        baseURL: 'http://localhost:3000',
+      },
+      nodeConnectionConfig,
+    );
+
+    // Creates a local or default provider
+    this.provider = web3
+      ? new ethers.providers.Web3Provider(web3.currentProvider)
+      : new ethers.providers.JsonRpcProvider({ url: urlProvider });
+  }
+
+  /**
+   * Initialize the module. Does nothing, exists only to implement IDataAccess
+   *
+   * @returns nothing
+   */
+  public async initialize(): Promise<void> {
+    // no-op, nothing to do
+    return;
+  }
+
+  /**
+   * Persists a new transaction on a node through HTTP.
+   *
+   * @param transactionData The transaction data
+   * @param topics The topics used to index the transaction
+   */
+  public async persistTransaction(
+    transactionData: DataAccessTypes.ITransaction,
+    channelId: string,
+    topics?: string[],
+  ): Promise<DataAccessTypes.IReturnPersistTransaction> {
+    if (!this.submitterContract) {
+      const network = await this.provider.getNetwork();
+
+      this.networkName =
```

Comment:
not used outside of this function, shouldn't be a class field

Reply:
It avoids to retreive the network information everytime you call the function. but rather do it once for good.

Reply:
hum, it's a very straightforward condition, not really an optimization... having it as a field, we might want to use it in other methods, where it would not be initialized.